src/main_character.cpp

```diff
@@ -1,15 +1,45 @@
 #include "character.hpp"
 #include "main_character.hpp"
 #include "sprite_sheet.hpp"
+#include <math.h>
+#include <iostream>
 
-
-MainCharacter::MainCharacter(const glob::vect &pos) : Character(pos, { 30, 30 }) {
+/**
+ * @brief Construct a new Main Character:: Main Character object
+ * Has default velocity of <30, 30>
+ * @param pos Position to set MainCharacter to 
+ */
+MainCharacter::MainCharacter(const glob::vect &pos) : Character(pos, {30, 30})
+{
 	_sprite_sheet.set_loop(_walking);
-	_speed = 200;
+	_speed = 500;
 }
 
+void MainCharacter::update()
+{
+	glob::vect unitDirVectors(0, 0);
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::Up))
+	{
+		unitDirVectors += glob::vect(0, 1);
+	}
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::Down))
+	{
+		unitDirVectors += glob::vect(0, -1);
+	}
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::Left))
+	{
+		unitDirVectors += glob::vect(-1, 0);
+	}
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::Right))
+	{
+		unitDirVectors += glob::vect(1, 0);
+	}
+	_velocity = glob::normalize(unitDirVectors) * _speed;
+	_pos += _velocity * glob::dt;
+}
```

Comment:
This should not happen inside of main character. Entities shouldn't control themselves, they leave that to a view. In the case of user_view, we talked about maybe removing it and  putting it's tasks into the handle_event method in state_playing, which would be okay.